orocos_kdl/src/chainidsolver_vereshchagin.cpp

```diff
@@ -43,39 +47,52 @@ ChainIdSolver_Vereshchagin::ChainIdSolver_Vereshchagin(const Chain& chain_, Twis
     Vm = MatrixXd::Identity(nc, nc);
     Sm = VectorXd::Ones(nc);
     tmpm = VectorXd::Ones(nc);
+
+    // Provide the necessary memory for storing joint effective inertia
+    d.resize(nj);
+
+    // Extract joint effective inertia from the model and store it in the allocated variable
+    int j = nj - 1;
+    for (int i = ns - 1; i >= 0; i--)
+    {
+        if (chain.getSegment(i).getJoint().getType() != Joint::Fixed)
+        {
+            d(j) = chain.getSegment(i).getJoint().getInertia();
+            j--;
```

Comment:
This means that the first few elements will be empty in case of fixed joints. Is there a more elegant solution possible?

Reply:
The number of elements in the joint rotor inertia ("d" ) variable, i.e. size of it, should only correspond to the number of active (not-fixed) joint. The reasons for that are the following:
* It does not make sense to consider (include in computations) inertia of the joints that do not perform any motion. If we do that, we might corrupt the accuracy of the solver, since the joint rotor inertia is added in the overall inertia variable D.
* In the "downwards_sweep" of the solver, authors (Ruben and Azamat) made the same indexing for the joints. See, for example how the feedforward joint torques are added in 238 line.

Reply:
Yes. A new, more elegant solution is implemented in a new commit.